admin-settings: add note regarding KinD + ECI config. (#22435)
docker/docs: content/manuals/security/for-admins/hardened-desktop/settings-management/configure-json-file.md

| &nbsp; &nbsp; &nbsp; &nbsp;`dockerDaemonOptions` |  | Overrides the options in the Linux daemon config file. See the [Docker Engine reference](/reference/cli/dockerd/#daemon-configuration-file).|  |

> [!NOTE]
>

> > This setting is not available to configure via the Docker Admin Console.

Reviewer: it isn't quite clear which setting this is referring to - Kubernetes or File sharing and emulation?

Author: In the deploy preview it shows up clearly under the Kubernetes section:

![image](https://github.com/user-attachments/assets/7aabfe1c-8686-44bd-be17-f016f0f70e18)

Author: That particular line is for the Windows Containers setting; I didn't add that line in this PR, my editor simply removed an extra whitespace in an empty line.

|:-------------------------------|---|:-------------------------------|---|
|`kubernetes`|  | If `enabled` is set to true, a Kubernetes single-node cluster is started when Docker Desktop starts. If `showSystemContainers` is set to true, Kubernetes containers are displayed in the Docker Desktop Dashboard and when you run `docker ps`.  `imagesRepository` lets you specify which repository Docker Desktop pulls the Kubernetes images from. For example, `"imagesRepository": "registry-1.docker.io/docker"`.  |  |

> [!NOTE]
>

> > When using a custom image repository via the `imagesRepository` setting, and if Enhanced Container Isolation (ECI) is enabled, add the following images to the [ECI Docker socket mount image list](#enhanced-container-isolation):
> >
> > `<custom-image-repo>/desktop-cloud-provider-kind:*`
> > `<custom-image-repo>/desktop-containerd-registry-mirror:*`
> >
> > The containers based on these images mount the Docker socket, so the images must be added to the ECI images list as otherwise ECI will block the mount and Kubernetes will fail to start.

Reviewer: ```suggestion
> When using the `imagesRepository` setting and Enhanced Container Isolation (ECI), add the following images to the [ECI Docker socket mount image list](#enhanced-container-isolation):
>
> `<custom-image-repo>/desktop-cloud-provider-kind:*`
> `<custom-image-repo>/desktop-containerd-registry-mirror:*`
>
> These containers mount the Docker socket, so you must add the images to the ECI images list. If not, ECI will block the mount and Kubernetes won't start.
```

rewrite suggestion for clarity
